plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity.java.ftl

```diff
@@ -604,17 +604,21 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
     </#if>
 
 	<#if data.waterMob>
-	@Override public boolean canBreatheUnderwater() {
-    	return true;
-    }
-
-    @Override public boolean checkSpawnObstruction(LevelReader world) {
+	@Override public boolean checkSpawnObstruction(LevelReader world) {
 		return world.isUnobstructed(this);
 	}
+	</#if>
 
-    @Override public boolean isPushedByFluid() {
-		return false;
-    }
+	<#if data.breatheUnderwater?? && (hasProcedure(data.breatheUnderwater) || data.breatheUnderwater.getFixedValue())>
+	@Override public boolean canBreatheUnderwater() {
+		return <@procedureOBJToConditionCode data.breatheUnderwater true false/>;
+	}
+	</#if>
+
+	<#if data.pushedByFluids?? && (hasProcedure(data.pushedByFluids) || !data.pushedByFluids.getFixedValue())>
+	@Override public boolean isPushedByFluid() {
+		return <@procedureOBJToConditionCode data.pushedByFluids false false/>;

```

Comment:
Both procedures lack dependency variables defined (meaning if a procedure with dependencies is used, the entity won't compile).

Reply:
Good catch, thank you for the review!